pkg/vsphere/tasks/waiter.go

```diff
@@ -185,13 +185,89 @@ func IsRetryError(op trace.Operation, err error) bool {
 	}
 }
 
+// IsConcurrentAccessError checks if a soap fault or vim fault is ConcurrentAccess error
+func IsConcurrentAccessError(err error) bool {
+	if soap.IsVimFault(err) {
+		_, ok := soap.ToVimFault(err).(*types.ConcurrentAccess)
+
+		return ok
+	}
+
+	if soap.IsSoapFault(err) {
+		_, ok := soap.ToSoapFault(err).VimFault().(types.ConcurrentAccess)
+
+		// sometimes we get the correct fault but wrong type
```

Comment:
Do you have some examples of this? We should raise a vSphere bug if they're returning the wrong or inconsistent types. If it's happening during govmomi deserialization then we should definitely let @dougm know.

Reply:
Ah - looks like this was a comment by @caglar10ur so we'll just let @dougm know for follow up if it's something he's unaware of.

Reply:
Yes it's an old comment made by @caglar10ur a while back. I didn't see the same behavior he described in the comment and I'm not sure under what conditions the behavior is likely to occur...